src/Common/ZooKeeper/ZooKeeper.cpp

```diff
@@ -771,7 +771,15 @@ ZooKeeper::multiImpl(const Coordination::Requests & requests, Coordination::Resp
 
     if (future_result.wait_for(std::chrono::milliseconds(args.operation_timeout_ms)) != std::future_status::ready)
     {
-        impl->finalize(fmt::format("Operation timeout on {} {}", Coordination::OpNum::Multi, requests[0]->getPath()));
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wpotentially-evaluated-expression"
+        impl->finalize(fmt::format(
+            "Operation timeout on {} of {} requests. First ({}): {}",
+            Coordination::OpNum::Multi,
+            requests.size(),
+            demangle(typeid(*requests[0]).name()),
+            requests[0]->getPath()));
+#pragma clang diagnostic pop
```

Comment:
maybe let's avoid tinkering clang warnings in the code?
```suggestion
        auto & request = *requests[0];
        impl->finalize(fmt::format(
            "Operation timeout on {} of {} requests. First ({}): {}",
            Coordination::OpNum::Multi,
            requests.size(),
            demangle(typeid(request).name()),
            request.getPath()));
```